src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,60 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidInputDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int REQUIRED_LENGTH = 6;
+    private static final int REQUIRED_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkForNullOrIncorrectInput(user);
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidInputDataException(
+                    "User with this login - " + user.getLogin()
+                            + " - already registered. Try to log in");
+        }
+        if (validateUserInfo(user)) {
+            storageDao.add(user);
+        }
+
+        return user;
+    }
+
+    private void checkForNullOrIncorrectInput(User user) {
+        if (user == null) {
+            throw new InvalidInputDataException("User can't be null!");
+        }
+        if (user.getLogin() == null) {
+            throw new InvalidInputDataException("Your login can't be null!");
+        }
+        if (user.getAge() <= 0) {
+            throw new InvalidInputDataException("Your age can't be less or equals zero!");
+        }
+        if (user.getPassword() == null) {
+            throw new InvalidInputDataException("Password can't be null!");
+        }
+    }
```

Comment:
TBH I don't like it a lot. I'd rather separate this into different methods.
For example
```suggestion
        private void checkUserAge(User user) {
        if (user.getAge() == null || user.getAge() < MIN_LEGAL_AGE) {
            throw new RegistrationFailureException("User's age is less than"
                    + MIN_LEGAL_AGE + " or not specified");
        }
    }
```